sdcflows/interfaces/bspline.py

```diff
@@ -192,6 +209,27 @@ def _run_interface(self, runtime):
             index += n
             self._results["out_coeff"].append(out_level)
 
+        # Interpolating in the original grid will require a new collocation matrix
+        if need_resize:
+            fmapnii = nb.load(self.inputs.in_data)
+            data = fmapnii.get_fdata(dtype="float32")
+            mask = (
+                np.ones_like(fmapnii.dataobj, dtype=bool) if masknii is None
+                else np.asanyarray(nb.load(self.inputs.in_mask).dataobj) > 1e-4
+            )
+            colmat, _ = _collocation_matrix(fmapnii, bs_spacing)
```

Comment:
this block feels redundant - we can just calculate `orig_colmat` in the first `need_resize` block and then use it if we resized

Reply:
this postpones the generation of a second collocation matrix to the end. The new one is only generated if the input was resized. I don't think I follow the argument here.

Reply:
Also, it is a large CSR matrix, so probably you want it to be as short-lived as possible.